storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc

```diff
@@ -774,7 +774,8 @@ bool xbcloud_download(Object_store *store, const std::string &container,
                       const std::string &backup_name) {
   std::vector<std::string> object_list;
 
-  if (!store->list_objects_in_directory(container, backup_name, object_list)) {
+  if (!store->list_objects_in_directory(container, backup_name, object_list) ||
```

Comment:
so with invalid backupname, list_objects_in_directory doesn't fail (ie doesn't return false?)

Reply:
no it won't return false as long as it has proper access on bucket.